app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -209,56 +208,12 @@ class ReportingRepositoryImpl(
                 quarantineRepository.markMissingExposureKeysAsUploaded()
             } else {
                 // The regular flow of reporting the exposure keys.
-                if (infectionLevel == MessageType.InfectionLevel.Red) {
-                    val sentTemporaryExposureKeys =
-                        infectionMessengerRepository.getSentTemporaryExposureKeysByMessageType(
-                            MessageType.InfectionLevel.Yellow
-                        )
-
-                    infectionMessages.addAll(
-                        sentTemporaryExposureKeys
-                            .map { message ->
-                                TemporaryExposureKeysWrapper(
-                                    message.rollingStartIntervalNumber,
-                                    message.password,
-                                    MessageType.InfectionLevel.Red
-                                )
-                            }
-                    )
-
-                    if (infectionMessages.isNotEmpty()) {
-                        infectionMessages.sortByDescending { content -> content.rollingStartIntervalNumber }
-                        thresholdTime = infectionMessages.first().rollingStartIntervalNumber
-                    }
-                } else if (infectionLevel == MessageType.InfectionLevel.Yellow) {
-                    val resetMessages =
-                        infectionMessengerRepository.getSentTemporaryExposureKeysByMessageType(
-                            MessageType.InfectionLevel.Yellow
-                        )
-                            .map { temporaryExposureKey ->
-                                TemporaryExposureKeysWrapper(
-                                    temporaryExposureKey.rollingStartIntervalNumber,
-                                    temporaryExposureKey.password,
-                                    MessageType.Revoke.Suspicion
-                                )
-                            }
-
-                    infectionMessengerRepository.storeSentTemporaryExposureKeys(resetMessages)
+                val infectionMessages = if (infectionLevel == MessageType.InfectionLevel.Red) {
+                    prepareListOfTemporaryExposureKeysForRedReporting(temporaryExposureKeysFromSDK, thresholdTime)
+                } else {
+                    prepareListOfTemporaryExposureKeysForYellowReporting(temporaryExposureKeysFromSDK, thresholdTime)
                 }
 
-                infectionMessages.addAll(
-                    temporaryExposureKeysFromSDK
-                        .filter { it.rollingStartIntervalNumber > thresholdTime }
-                        .groupBy { it.rollingStartIntervalNumber }
-                        .map { (rollingStartIntervalNumber, _) ->
-                            TemporaryExposureKeysWrapper(
-                                rollingStartIntervalNumber,
-                                UUID.randomUUID(),
```

Comment:
Sure the key has not yet been used? i.e. does not have a password associated yet?

Reply:
This logic has been moved in:
-  `prepareListOfTemporaryExposureKeysForRedReporting` where we generate passwords for the newer keys in the case when we do the reporting after a yellow reporting
- `prepareListWithGreenTemporaryExposureKeysFromDbAndNewKeysFromTheSDK` 
      - where we generate password when we do a clean submission, no keys at all in db.
      - where we generate password for newer keys than the the green keys that we have in db.